src/tbb/arena_slot.cpp

```diff
@@ -142,12 +145,23 @@ d1::task* arena_slot::get_task(execution_data_ext& ed, isolation_type isolation)
 
     __TBB_ASSERT( (std::intptr_t)tail.load(std::memory_order_relaxed) >= 0, nullptr );
     __TBB_ASSERT( result || tasks_omitted || is_quiescent_local_task_pool_reset(), nullptr );
+#ifdef __TBB_STATISTICS
+    if (result) {
+        governor::get_thread_data()->add_statistics(thread_data::Statistics::TASK_RES_NOT_NULLPTR);
+    }
+#endif /* __TBB_STATISTICS */
     return result;
 }
 
 d1::task* arena_slot::steal_task(arena& a, isolation_type isolation) {
+#ifdef __TBB_STATISTICS
+    governor::get_thread_data()->add_statistics(thread_data::Statistics::TRY_STEAL_TASK);
+#endif /* __TBB_STATISTICS */
     d1::task** victim_pool = lock_task_pool();
     if (!victim_pool) {
+#ifdef __TBB_STATISTICS
+        governor::get_thread_data()->add_statistics(thread_data::Statistics::VICTIM_POOL_EMPTY);
+#endif /* __TBB_STATISTICS */
```

Comment:
Taking TLS (`governor::get_thread_data()`) is not cheapest operation, in many places `thread_data` is available (or consider passing it trough the function arguments).  @pavelkumbrasev , what do you think?